pkg/tsdb/prometheus/querydata/request.go

```diff
@@ -118,62 +118,68 @@ func (s *QueryData) fetch(ctx context.Context, client *client.Client, q *models.
 	logger := s.log.FromContext(traceCtx)
```

Comment:
`func (s *QueryData) fetch(ctx context.Context, client *client.Client, q *models.Query, headers map[string]string) (*backend.DataResponse, error)`

Since this doesn't return an error, I believe the signature can change to `func (s *QueryData) fetch(ctx context.Context, client *client.Client, q *models.Query, headers map[string]string) (*backend.DataResponse)` (remove error)